src/system/unix/network.c

```diff
@@ -377,10 +377,33 @@ void *_z_listen_udp_multicast(void *arg, uint32_t tout, const char *iface) {
 
     // Join the multicast group
     if (raddr->ai_family == AF_INET) {
+        struct ifaddrs *l_ifaddr = NULL;
+        if (getifaddrs(&l_ifaddr) < 0) {
+            goto _Z_LISTEN_UDP_MULTICAST_ERROR_2;
+        }
+
+	unsigned long l_iface_ip = htonl(INADDR_ANY);
+        struct ifaddrs *tmp = NULL;
+        for (tmp = l_ifaddr; tmp != NULL; tmp = tmp->ifa_next) {
+            if (_z_str_eq(tmp->ifa_name, iface)) {
+                if (tmp->ifa_addr->sa_family == raddr->ai_family) {
+                    if (tmp->ifa_addr->sa_family == AF_INET) {
+                        l_iface_ip = ((struct sockaddr_in*)tmp->ifa_addr)->sin_addr.s_addr;
+                    }
+                    else {
+                        continue;
+                    }
+
+                    break;
+                }
+            }
+        }
+        freeifaddrs(l_ifaddr);
+
```

Comment:
I suggest to move the newly-added block of code to a dedicated function and not part of the logic.

Reply:
Converted into an utility function.